### PR TITLE
feat(block, block-section): adds status and relate UI 

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #2261

## Summary
This adds status to Block and BlockSection.

### CODE REVIEW

File: src/components/calcite-block-section/calcite-block-section.tsx [hunk 1]
```diff
@@ -92,6 +98,19 @@ export class CalciteBlockSection {
...
+    const { status } = this;
+    const statusIcon = ICONS[status] ?? false;
+    const statusIconClasses = {
+      [CSS.statusIcon]: true,
+      [CSS.valid]: status == "valid",
+      [CSS.invalid]: status == "invalid"
+    };
+
+    return !!statusIcon ? (
+      <calcite-icon class={statusIconClasses} icon={statusIcon} scale="s"></calcite-icon>
```
Comment (driskull) [inline src/components/calcite-block-section/calcite-block-section.tsx:111]: nitpick. self close the `calcite-icon`. `<calcite-icon class={statusIconClasses} icon={statusIcon} scale="s" />`

File: src/components/calcite-block-section/calcite-block-section.tsx [hunk 2]
```diff
@@ -104,6 +123,8 @@ export class CalciteBlockSection {
 
     const toggleLabel = open ? intlCollapse || TEXT.collapse : intlExpand || TEXT.expand;
 
+    
+
```
Comment (driskull) [inline src/components/calcite-block-section/calcite-block-section.tsx:127]: spacing. run prettier :)

**Review by driskull — APPROVED** [review]
👍

File: src/components/calcite-block-section/calcite-block-section.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-block-section/calcite-block-section.tsx]: fixed.

File: src/components/calcite-block-section/calcite-block-section.tsx (hunk 2 above)
Comment (asangma) [reply src/components/calcite-block-section/calcite-block-section.tsx]: done

**Comment by macandcheese** [comment]
Have these designs been shared and reviewed with @bstifle or anybody else ?

**Comment by bstifle** [comment]
I haven't seen these, @asangma do you have some screenshots in here? Nothing in the demo html page. Not up to speed on the use case either :)

**Comment by bstifle** [comment]
nevermind, saw the other issue.

Is there a build to look at?

**Comment by asangma** [comment]
@bstifle You can build this branch or we can review together.

**Comment by asangma** [comment]
> Have these designs been shared and reviewed with @bstifle or anybody else ?

Anybody else? Yes. Probably 3-7 anybody elses.

**Comment by asangma** [comment]
Thanks @bstifle.

**Comment by bstifle** [comment]
Ship it! looks really good

**Comment by asangma** [comment]
:shipit: 
Just waiting on checks.

File: src/components/calcite-block-section/calcite-block-section.tsx (hunk 2 above)
Comment (jcfranco) [reply src/components/calcite-block-section/calcite-block-section.tsx]: Weird. Linting should run automatically when you commit. @asangma Do you see something like the following when you commit?

```zsh
~/.../calcite-components my-branch* ⇣
❯ git commit -m 'chore: do stuff'
husky > pre-commit (node v14.17.0)
✔ Preparing...
✔ Running tasks...
✔ Applying modifications...
✔ Cleaning up...
[my-branch 102c64930] chore: do stuff
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 some-file.json
```

**Pull request merged**